Framework/Foundation/include/Framework/Pack.h

```diff
@@ -90,6 +90,15 @@ constexpr auto concatenate_pack(P1 p1, P2 p2, Ps... ps)
 template <typename... Ps>
 using concatenated_pack_t = decltype(concatenate_pack(Ps{}...));
 
+template <typename... Args1, typename... Args2>
+constexpr auto interleave_pack(pack<Args1...>, pack<Args2...>)
+{
+  return concatenated_pack_t<pack<Args1, Args2>...>{};
+}
+
+template <typename... Ps>
+using interleaved_pack_t = decltype(interleave_pack(Ps{}...));
+
```

Comment:
Thanks. It would be good if you moved the Pack.h changes in a separate PR and add a couple of tests to show the newly introduced  features.

Reply:
Same here. Please, address the comment about `pack_to_tuple` and the we discuss again.